File: examples/v2/environment-config/python/__main__.py
import pulumi
import pulumi_pulumiservice.v2 as ps_v2

config = pulumi.Config()
organization_name = config.get("organizationName") or "service-provider-test-org"
project_name = config.get("projectName") or "v2-envcfg-example"
env_name = config.get("envName") or "v2-envcfg-env"

draft = ps_v2.esc.EnvironmentDraft(
    "draft",
    org_name=organization_name,
    project_name=project_name,
    env_name=env_name,
)

settings = ps_v2.esc.EnvironmentSettings(
    "settings",
    org_name=organization_name,
    project_name=project_name,
    env_name=env_name,
    deletion_protected=True,
)

pulumi.export("draftId", draft.change_request_id)
pulumi.export("protected", settings.deletion_protected)

File: examples/v2/environment-config/go/main.go
package main

import (
	esc "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v2/esc"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		serviceOrg := cfg.Get("serviceOrg")
		if serviceOrg == "" {
			serviceOrg = "service-provider-test-org"
		}
		projectName := cfg.Get("projectName")
		if projectName == "" {
			projectName = "v2-envcfg-example"
		}
		envName := cfg.Get("envName")
		if envName == "" {
			envName = "v2-envcfg-env"
		}

		draft, err := esc.NewEnvironmentDraft(ctx, "draft", &esc.EnvironmentDraftArgs{
			OrgName:     pulumi.String(serviceOrg),
			ProjectName: pulumi.String(projectName),
			EnvName:     pulumi.String(envName),
		})
		if err != nil {
			return err
		}

		settings, err := esc.NewEnvironmentSettings(ctx, "settings", &esc.EnvironmentSettingsArgs{
			OrgName:           pulumi.String(serviceOrg),
			ProjectName:       pulumi.String(projectName),
			EnvName:           pulumi.String(envName),
			DeletionProtected: pulumi.Bool(true),
		})
		if err != nil {
			return err
		}

		ctx.Export("draftId", draft.ChangeRequestID)
		ctx.Export("protected", settings.DeletionProtected)
		return nil
	})
}
